safe_csv.go

```diff
@@ -2,16 +2,22 @@ package gocsv
 
 //Wraps around SafeCSVWriter and makes it thread safe.
 import (
-	"encoding/csv"
 	"sync"
 )
 
 type SafeCSVWriter struct {
-	*csv.Writer
+	Writer CSVWriter
 	m sync.Mutex
 }
 
-func NewSafeCSVWriter(original *csv.Writer) *SafeCSVWriter {
+type CSVWriter interface {
+	Error() error
+	Flush()
+	Write(record []string) error
+	WriteAll(records [][]string) error
+}
```

Comment:
You might not actually need all these functions, since SafeCSVWriter doesn't end up being a drop-in replacement for csv.Writer anyway (since it has the Comma and UseCRLF fields). Maybe this can just be Write(), since that's all that's needed here.